core/types/legacy_tx.go

```diff
@@ -67,6 +67,22 @@ type LegacyTx struct {
 func (tx LegacyTx) GetPrice() *uint256.Int  { return tx.GasPrice }
 func (tx LegacyTx) GetTip() *uint256.Int    { return tx.GasPrice }
 func (tx LegacyTx) GetFeeCap() *uint256.Int { return tx.GasPrice }
+func (tx LegacyTx) GetEffectiveGasTip(baseFee *uint256.Int) *uint256.Int {
+	if baseFee == nil {
+		return tx.GetTip()
+	}
+	gasFeeCap := tx.GetFeeCap()
+	// return 0 because effectiveFee cant be < 0
+	if gasFeeCap.Cmp(baseFee) == -1 {
```

Comment:
Just a small nitpick, `uint256.Int` has more convinient functions `Lt` and `Gt` and `Eq` :)

Reply:
Nice, I've changed to using that.